subsystem/ams_subscriptions_test.go

```diff
@@ -343,12 +343,15 @@ var _ = Describe("test AMS subscriptions", func() {
 			By("update subscription with openshfit (external) cluster ID", func() {
 				// in order to simulate infra env generation
 				generateClusterISO(clusterID, models.ImageTypeMinimalIso)
-				registerHostsAndSetRoles(clusterID, minHosts)
+				registerHostsAndSetRoles(clusterID, minHosts, "test-ams-subscriptions-cluster", "example.com")
 				reply, err = userBMClient.Installer.InstallCluster(context.Background(), &installer.InstallClusterParams{ClusterID: clusterID})
 				Expect(err).NotTo(HaveOccurred())
 				c := reply.GetPayload()
 				Expect(*c.Status).Should(Equal(models.ClusterStatusPreparingForInstallation))
 				generateEssentialPrepareForInstallationSteps(ctx, c.Hosts...)
+				for _, host := range c.Hosts {
+					updateProgress(*host.ID, clusterID, models.HostStageDone)
```

Comment:
This cannot work, the test title is `UpdateSubscription failed`.

You are trying to simulate the host's installation on a cluster that is failing to install,  you can look on how it is done in `happy flow` test - https://github.com/openshift/assisted-service/blob/dd818a5650a66ff6693f632ddb83d48d29692755/subsystem/ams_subscriptions_test.go#L310

Reply:
you right, thanks! I have been trying to understand why it keeps failing, removing these lines.